File: sdks/dotnet/Sequins.OpenTelemetry.Tests/SequinsTests.cs
using Sequins.OpenTelemetry;
using Xunit;

namespace Sequins.OpenTelemetry.Tests;

public class SequinsTests
{
    [Fact]
    public void Build_ReturnsHandle()
    {
        using var handle = new SequinsBuilder("test-service").Build();
        Assert.NotNull(handle);
    }

    [Fact]
    public void TracerProvider_IsNotNull()
    {
        using var handle = new SequinsBuilder("test-tracer").Build();
        Assert.NotNull(handle.TracerProvider);
    }

    [Fact]
    public void MeterProvider_IsNotNull()
    {
        using var handle = new SequinsBuilder("test-meter").Build();
        Assert.NotNull(handle.MeterProvider);
    }

    [Fact]
    public void LoggerProvider_IsNotNull()
    {
        using var handle = new SequinsBuilder("test-logger").Build();
        Assert.NotNull(handle.LoggerProvider);
    }

    [Fact]
    public void Dispose_DoesNotThrow()
    {
        var handle = new SequinsBuilder("test-dispose").Build();
        var ex = Record.Exception(() => handle.Dispose());
        Assert.Null(ex);
    }

    [Fact]
    public void CustomOptions_Applied()
    {
        var options = new SequinsOptions
        {
            ServiceName = "custom-service",
            Endpoint = "http://localhost:4317",
        };
        using var handle = new SequinsBuilder(options).Build();
        Assert.NotNull(handle);
    }
}

File: sdks/swift/Tests/SequinsOtelTests/SequinsTests.swift
import XCTest
@testable import SequinsOtel

final class SequinsTests: XCTestCase {
    func testInitializeReturnsHandle() throws {
        let handle = try Sequins.initialize(serviceName: "test-service")
        XCTAssertNotNil(handle)
        handle.shutdown()
    }

    func testTracerProviderIsNotNil() throws {
        let handle = try Sequins.initialize(serviceName: "test-tracer")
        XCTAssertNotNil(handle.tracerProvider)
        handle.shutdown()
    }

    func testShutdownDoesNotCrash() throws {
        let handle = try Sequins.initialize(serviceName: "test-shutdown")
        handle.shutdown()
    }

    func testCustomConfig() throws {
        let config = SequinsConfig(
            serviceName: "custom-service",
            endpoint: "http://localhost:4318"
        )
        let handle = try Sequins.initialize(config: config)
        XCTAssertNotNil(handle)
        handle.shutdown()
    }
}
